packages/commercetools/api-client/src/helpers/cart/actions.ts

```diff
@@ -58,9 +58,17 @@ export const setShippingAddressAction = (shippingDetails: Address): { setShippin
         building: shippingDetails.building,
         apartment: shippingDetails.apartment,
         pOBox: shippingDetails.pOBox,
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
ignores? why?

Reply:
`contactInfo` was removed from `Address`. We already logged a warning if somebody used it in projects (lines 38-40).

However, I want this PR to be released as a patch, and removing `|| shippingDetails.contactInfo?.X` would be a breaking change. That's why in here, I added `@ts-ignore`, but in separate PR for `develop` (next minor release), I will remove this completely, together with a deprecation warning.

Reply:
```ts
// this you import
type AddressBase = {
  something: string
}

// comment here as legacy
type Address = AddressBase & {
  contactInfo: {
    street: string
  }
}


const testFunction = (shippingDetails: Address) => {

  return {
    test: shippingDetails.something,
    street: shippingDetails?.contactInfo.street || 'test'
  }
}
```

Reply:
or edit generated file, and just add those fields in the interface/type but with the comment that's deprecated

Reply:
however, this is BC break that already happened - if someone will access contactInfo field, it will raise an error it get undefined data even if you still saying (from the TS perspective that still exist)

Reply:
what's more... you are saing that ts-ignores are for preventing from BC's while when you generated a new types, they don't have contactInfo so... means BC